src/middleware/condition.rs

```diff
@@ -0,0 +1,143 @@
+use actix_service::{Service, Transform};
+use futures::future::{ok, Either, FutureResult, Map};
+use futures::{Future, Poll};
+
+/// `Middleware` for conditianally enables another middleware
+/// The controled middleware must not change the `Service` interfaces.
+/// This means you cannot control such middlewares like `Logger` or `Compress`.
+///
+/// ## Usage
+///
+/// ```
```

Comment:
IMO: We have explicit `rust` annotation in other places so it seems reasonable to add it here.